Support multiple boms at the starting DEPS

Sometimes we might want to use multiple boms in a single script...

```java
///usr/bin/env jbang "$0" "$@" ; exit $?
//DEPS io.vertx:vertx-stack-depchain:4.0.3@pom
//DEPS org.apache.camel:camel-bom:3.9.0@pom
//DEPS io.vertx:vertx-core
//DEPS org.apache.camel:camel-core
//DEPS org.apache.camel:camel-stream
//DEPS org.apache.camel:camel-vertx
//DEPS org.slf4j:slf4j-simple:1.7.30
...
```

```console
$ jbang camel-vertx.java 
[jbang] Resolving dependencies...
[jbang] Loading MavenCoordinate [io.vertx:vertx-stack-depchain:pom:4.0.3]
[jbang] [ERROR] POM imports as found in org.apache.camel:camel-bom:3.9.0@pom is only supported as the first import.
[jbang] Run with --verbose for more details
```

--- a/src/main/java/dev/jbang/dependencies/DependencyUtil.java
+++ b/src/main/java/dev/jbang/dependencies/DependencyUtil.java
@@ -6,6 +6,9 @@ import static dev.jbang.util.Util.infoHeader;
 import static dev.jbang.util.Util.infoMsg;
 import static dev.jbang.util.Util.infoMsgFmt;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
@@ -32,6 +35,9 @@ import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenCoordinates;
 
 import dev.jbang.Settings;
 import dev.jbang.cli.ExitException;
+import dev.jbang.util.Util;
+
+import picocli.CommandLine;
 
 public class DependencyUtil {
 
@@ -149,24 +155,57 @@ public class DependencyUtil {
 
 		System.setProperty("maven.repo.local", Settings.getLocalMavenRepo().toPath().toAbsolutePath().toString());
 
-		List<MavenCoordinate> coords = depIds	.stream()
-												.map(DependencyUtil::depIdToArtifact)
-												.collect(Collectors.toList());
+		Map<Boolean, List<MavenCoordinate>> coordList = depIds	.stream()
+																.map(DependencyUtil::depIdToArtifact)
+																.collect(Collectors.partitioningBy(
+																		c -> c.getType().equals(PackagingType.POM)));
+
+		List<MavenCoordinate> coords = coordList.get(false);
+		List<MavenCoordinate> pomcoords = coordList.get(true);
 
 		PomEquippedResolveStage pomResolve = null;
-		if (!coords.isEmpty()) {
-			MavenCoordinate mc = coords.get(0);
-			if (PackagingType.POM.equals(mc.getType())) {
-				if (loggingEnabled) {
-					infoMsg("Fetching POM: " + mc.toCanonicalForm());
-				}
-				System.setProperty("jbang-allowpom", "true"); // big hack to trick shrinkwrap in actually get pom
-																// location
-				MavenStrategyStage resolve = resolver.resolve(mc.toCanonicalForm());
-				pomResolve = resolver.loadPomFromFile(resolve.withoutTransitivity().asSingleFile());
-				System.getProperties().remove("jbang-allowpom");
-				coords.remove(0);
+		if (!pomcoords.isEmpty()) {
+
+			String beforeDepMgmt = "<project xmlns=\"http://maven.apache.org/POM/4.0.0\"\n" +
+					"         xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\"\n" +
+					"         xsi:schemaLocation=\"http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd\">\n"
+					+
+					"    <modelVersion>4.0.0</modelVersion>\n" +
+					"\n" +
+					"    <groupId>dev.jbang.internal</groupId>\n" +
+					"    <artifactId>dependency-pom</artifactId>\n" +
+					"    <version>1.0-SNAPSHOT</version>\n" +
+					"    <packaging>pom</packaging>\n" +
+					"<dependencyManagement>\n" +
+					"    <dependencies>\n";
+			String afterDepMgmt = "</dependencies>\n" +
+					"</dependencyManagement>\n" +
+					"</project>";
+			StringBuffer buf = new StringBuffer(beforeDepMgmt);
+			for (MavenCoordinate pomcoord : pomcoords) {
+				buf.append("<dependency>\n" +
+						"      <groupId>" + pomcoord.getGroupId() + "</groupId>\n" +
+						"      <artifactId>" + pomcoord.getArtifactId() + "</artifactId>\n" +
+						"      <version>" + pomcoord.getVersion() + "</version>\n" +
+						"             <type>pom</type>\n" +
+						"      <scope>import</scope>\n" +
+						"    </dependency>\n");
 			}
+			buf.append(afterDepMgmt);
+			Path pompath = null;
+			try {
+				pompath = File.createTempFile("jbang", ".xml").toPath();
+				Util.writeString(pompath, buf.toString());
+			} catch (IOException e) {
+				throw new ExitException(CommandLine.ExitCode.SOFTWARE,
+						"Error trying to generate pom.xml for dependency management");
+			}
+			if (loggingEnabled) {
+				infoMsg("Artifacts used for dependency management:");
+				infoMsgFmt("         %s\n", String.join("\n         ",
+						pomcoords.stream().map(Coordinate::toCanonicalForm).collect(Collectors.toList())));
+			}
+			pomResolve = resolver.loadPomFromFile(pompath.toFile());
 		}
 
 		Optional<MavenCoordinate> pom = coords.stream().filter(c -> c.getType().equals(PackagingType.POM)).findFirst();

--- a/src/test/java/dev/jbang/dependencies/DependencyResolverTest.java
+++ b/src/test/java/dev/jbang/dependencies/DependencyResolverTest.java
@@ -14,6 +14,7 @@ import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
+import java.util.Optional;
 import java.util.Properties;
 
 import org.jboss.shrinkwrap.resolver.api.maven.coordinate.MavenCoordinate;
@@ -100,8 +101,7 @@ class DependencyResolverTest extends BaseTest {
 				PropertiesValueResolver.replaceProperties("org.openjfx:javafx-base:11.0.2:${os.detected.jfxname}", p));
 
 		List<ArtifactInfo> artifacts = DependencyUtil.resolveDependenciesViaAether(deps,
-				Collections.singletonList(toMavenRepo("mavencentral")), false,
-				true, true);
+				Collections.singletonList(toMavenRepo("mavencentral")), false, true, true);
 
 		assertEquals(1, artifacts.size());
 	}
@@ -111,8 +111,7 @@ class DependencyResolverTest extends BaseTest {
 		List<String> deps = Arrays.asList("com.offbytwo:docopt:0.6.0.20150202", "log4j:log4j:1.2+");
 
 		List<ArtifactInfo> artifacts = DependencyUtil.resolveDependenciesViaAether(deps,
-				Collections.singletonList(toMavenRepo("mavencentral")), false,
-				true, true);
+				Collections.singletonList(toMavenRepo("mavencentral")), false, true, true);
 
 		assertEquals(2, artifacts.size());
 	}
@@ -136,8 +135,7 @@ class DependencyResolverTest extends BaseTest {
 
 	@Test
 	void testResolveDependenciesNoDuplicates() {
-		List<String> deps = Arrays.asList(
-				"org.apache.commons:commons-configuration2:2.7",
+		List<String> deps = Arrays.asList("org.apache.commons:commons-configuration2:2.7",
 				"org.apache.commons:commons-text:1.8");
 
 		ModularClassPath classpath = DependencyUtil.resolveDependencies(deps, Collections.emptyList(), false, false,
@@ -195,13 +193,65 @@ class DependencyResolverTest extends BaseTest {
 	 */
 	@Test
 	void testImportPOM() {
-		List<String> deps = Arrays.asList("com.microsoft.azure:azure-bom:1.0.0.M1@pom",
-				"com.microsoft.azure:azure");
+		List<String> deps = Arrays.asList("com.microsoft.azure:azure-bom:1.0.0.M1@pom", "com.microsoft.azure:azure");
 
-		ModularClassPath classpath = DependencyUtil.resolveDependencies(deps,
-				Collections.emptyList(), false, false, true);
+		ModularClassPath classpath = DependencyUtil.resolveDependencies(deps, Collections.emptyList(), false, false,
+				true);
 
 		assertEquals(62, classpath.getArtifacts().size());
+	}
+
+	@Test
+	void testImportMultipleBoms() {
+		List<String> deps = Arrays.asList("io.vertx:vertx-stack-depchain:4.2.3@pom", // if not listed then vertx.core
+																						// will be version 3.9.5
+				"org.apache.camel:camel-bom:3.9.0@pom",
+				// "io.vertx:vertx-core",
+				"org.apache.camel:camel-core",
+				"org.apache.camel:camel-vertx",
+				"org.slf4j:slf4j-simple:1.7.30");
+
+		ModularClassPath classpath = DependencyUtil.resolveDependencies(deps, Collections.emptyList(), false, false,
+				true);
+
+		Optional<ArtifactInfo> coord = classpath.getArtifacts()
+												.stream()
+												.filter(ai -> ai.getCoordinate()
+																.toCanonicalForm()
+																.startsWith("io.vertx:vertx-core"))
+												.findFirst();
+
+		assertEquals(coord.get().getCoordinate().getVersion(), "4.2.3");
+
+		coord = classpath	.getArtifacts()
+							.stream()
+							.filter(ai -> ai.getCoordinate().toCanonicalForm().startsWith("org.slf4j:slf4j-simple:"))
+							.findFirst();
+
+		assertEquals(coord.get().getCoordinate().getVersion(), "1.7.30");
+
+		coord = classpath	.getArtifacts()
+							.stream()
+							.filter(ai -> ai.getCoordinate()
+											.toCanonicalForm()
+											.startsWith("org.apache.camel:camel-vertx"))
+							.findFirst();
+
+		assertEquals(coord.get().getCoordinate().getVersion(), "3.9.0");
+
+		deps = Arrays.asList(
+				"org.apache.camel:camel-bom:3.9.0@pom",
+				"org.apache.camel:camel-core",
+				"org.apache.camel:camel-vertx",
+				"org.slf4j:slf4j-simple:1.7.30");
+		classpath = DependencyUtil.resolveDependencies(deps, Collections.emptyList(), false, false, true);
+
+		coord = classpath	.getArtifacts()
+							.stream()
+							.filter(ai -> ai.getCoordinate().toCanonicalForm().startsWith("io.vertx:vertx-core"))
+							.findFirst();
+
+		assertEquals(coord.get().getCoordinate().getVersion(), "3.9.5");
 	}
 
 }

